pkg/trace/stats/concentrator_test.go

```diff
@@ -410,13 +489,161 @@ func TestConcentratorSublayersStatsCounts(t *testing.T) {
 		"query|hits|env:none,resource:resource4,service:A3":                                               1,
 		"query|hits|env:none,resource:resource6,service:A3":                                               1,
 	}
+	countValsEq(t, expectedCountValByKey, receivedCounts)
+}
+
+// setupConcentrator initializes a concentrator from a given trace input.
```

Comment:
The name of this function, versus what it actually does is completely off. This function does not set up a concentrator, it instead:
* Computes the top-level span of a trace
* Creates a new weighted trace
* Computes and extract sublayer statistics
* Creates a new concentrator
* Adds a trace to it
* Returns the concentrator.

It does a huge amount of things.

A unit test should be:
1. Have some input data
2. Call a function, method or initiated object
3. Assert an output

If we do more than the above, it is is a sign that:
* We are doing too much
* Our architecture is bad and we need to refactor to allow the above model

It's very likely that it is the latter, because this code is pretty old and warrants a refactor.

Either way, let's get rid of this function, and instead of having two tests, just have one test for the concentrator, since both your global variables (`addNowTests` and `sublayerMeasuredSpanTests `) are exactly the same, so just merge them and keep this code in the test.